plugins/module_utils/network/sonic/facts/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,321 @@
+                        self.update_dict(default_information_dict, "metric_type", config.get("metric-type"))
+                        if "metric_type" in default_information_dict:
+                            type_val = default_information_dict['metric_type'].split(":")[1]
+                            default_information_dict['metric_type'] = 1 if type_val == 'TYPE_1' else 2
+                        default_information_dict['originate'] = True
+                    else:
+                        self.update_dict(redistribute_dict, 'always', config.get("always"))
+                        self.update_dict(redistribute_dict, "metric", config.get("metric"))
+                        self.update_dict(redistribute_dict, "route_map", config.get("route-map"))
+                        self.update_dict(redistribute_dict, "metric_type", config.get("metric-type"))
+                        if "metric_type" in redistribute_dict:
+                            type_val = redistribute_dict['metric_type'].split(":")[1]
+                            redistribute_dict['metric_type'] = 1 if type_val == 'TYPE_1' else 2
+                        redistribute_dict['protocol'] = map_protocol
+            if redistribute_dict:
+                redistribute_list.append(redistribute_dict)
+
+        self.update_dict(return_dict, 'default_information', default_information_dict)
+        self.update_dict(return_dict, 'redistribute', redistribute_list)
+
+        return return_dict
+
+    def get_ospf_graceful_restart(self, graceful_restart):
+        return_dict = {}
+        config = graceful_restart.get("config")
+        if config:
+            self.update_dict(return_dict, 'graceful_restart_enable', config.get("enabled"))
+            self.update_dict(return_dict, 'grace_period', config.get("openconfig-ospfv2-ext:grace-period"))
+            helper_dict = {}
+            self.update_dict(helper_dict, 'enable', config.get("helper-only"))
+            self.update_dict(helper_dict, 'planned_only', config.get("openconfig-ospfv2-ext:planned-only"))
+            self.update_dict(helper_dict, 'strict_lsa_checking', config.get("openconfig-ospfv2-ext:strict-lsa-checking"))
+            self.update_dict(helper_dict, 'supported_grace_time', config.get("openconfig-ospfv2-ext:supported-grace-time"))
+            neighbor_id = []
+            helpers = graceful_restart.get('openconfig-ospfv2-ext:helpers', {})
+            for helper in helpers.get('helper', []):
+                if helper.get('neighbour-id'):
+                    neighbor_id.append(helper.get('neighbour-id'))
+            self.update_dict(helper_dict, 'neighbor_id', neighbor_id)
+            self.update_dict(return_dict, 'helper', helper_dict)
+
+        if return_dict:
+            return_dict = {'graceful_restart': return_dict}
+
+        return return_dict
+
+    def update_dict(self, dict, key, value, parent_key=None):
+        if value is not None and value not in [{}, [], ()]:
```

Comment:
Since json is parsed, the empty check can be limited.
```suggestion
        if value not in [None, {}, []]:
```